docs/program.rst

```diff
@@ -1,4 +1,38 @@
 Program
 =======
 
-contains Program details
\ No newline at end of file
+In the OpenQL programming model, one first creates the platform object and then with it the program object.
```

Comment:
It seems this is the closest one to a main introduction. Would it make sense to have this be before gates in the table of contents? 

The general programming model could also really use a figure on how these main concepts relate.